NitroxClient/MonoBehaviours/CinematicController/MultiplayerCinematicController.cs

```diff
@@ -0,0 +1,153 @@
+        }
+    }
+
+    private RemotePlayerCinematicController GetController(RemotePlayer player)
+    {
+        if (controllerByPlayerId.TryGetValue(player.PlayerId, out RemotePlayerCinematicController controller))
+        {
+            return controller;
+        }
+
+        player.PlayerDisconnectEvent.AddHandler(gameObject, OnPlayerDisconnect);
+
+        controller = CreateNewControllerForPlayer();
+        controllerByPlayerId.Add(player.PlayerId, controller);
+        return controller;
+    }
+
+    public void OnPlayerDisconnect(RemotePlayer player)
+    {
+        if (controllerByPlayerId.TryGetValue(player.PlayerId, out RemotePlayerCinematicController controller))
+        {
+            Destroy(controller);
+            controllerByPlayerId.Remove(player.PlayerId);
+        }
+    }
+
+    private RemotePlayerCinematicController CreateNewControllerForPlayer()
+    {
+        RemotePlayerCinematicController controller = gameObject.AddComponent<RemotePlayerCinematicController>();
+        controllerPrefab.PopulateRemoteController(controller);
+
+        return controller;
+    }
+
+    public void AddOtherControllers(IEnumerable<MultiplayerCinematicController> otherControllers)
+    {
+        foreach (MultiplayerCinematicController controller in otherControllers)
+        {
+            if (controller.playerController.animator == playerController.animator)
+            {
+                multiplayerControllerSameAnimator.Add(controller);
+            }
+        }
+    }
+
+    public static MultiplayerCinematicController Initialize(PlayerCinematicController playerController)
+    {
+        MultiplayerCinematicController mcp = playerController.gameObject.AddComponent<MultiplayerCinematicController>();
+        mcp.controllerPrefab = new CinematicControllerPrefab(playerController);
+        mcp.playerController = playerController;
+        return mcp;
+    }
+}
+
+public readonly struct CinematicControllerPrefab
+{
+    private readonly Transform animatedTransform;
+    private readonly GameObject informGameObject;
+    private readonly Transform endTransform;
+    private readonly bool onlyUseEndTransformInVr;
+    private readonly bool playInVr;
+    private readonly string playerViewAnimationName;
+    private readonly string playerViewInterpolateAnimParam;
+    private readonly string animParam;
+    private readonly string interpolateAnimParam;
+    private readonly float interpolationTime;
+    private readonly float interpolationTimeOut;
+    private readonly string receiversAnimParam;
+    private readonly GameObject[] animParamReceivers;
+    private readonly bool interpolateDuringAnimation;
+    private readonly Animator animator;
+
+    public CinematicControllerPrefab(PlayerCinematicController playerController)
+    {
+        animatedTransform = playerController.animatedTransform;
+        informGameObject = playerController.informGameObject; // TODO: This is sometimes set by on runtime. Currently not synced
```

Comment:
Is this something we are going to do in this pr or are we leaving it till later?

Reply:
This is something we will look into once issues pop up. So if testing is fine I will leave it till later (maybe it will never cause issue lol).

Reply:
Works for me, just as a note its applied here but the property isnt used to apply on the outgoing, im guessing thats as intended